src/mpid/ch4/netmod/ofi/ofi_coll.h

```diff
@@ -23,16 +23,27 @@
 static inline int MPIDI_NM_mpi_barrier(MPIR_Comm * comm_ptr, MPIR_Errflag_t * errflag,
                                        void *ch4_algo_parameters_container_in)
 {
-    int mpi_errno;
+    int mpi_errno = MPI_SUCCESS;
+    MPIDI_OFI_coll_algo_container_t *nm_algo_parameters_container_out = NULL;
+
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_NM_MPI_BARRIER);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_NM_MPI_BARRIER);
 
-    MPIDI_OFI_coll_algo_container_t *nm_algo_parameters_container_out = NULL;
-
     nm_algo_parameters_container_out =
         MPIDI_OFI_Barrier_select(comm_ptr, errflag, (MPIDI_OFI_coll_algo_container_t *)
                                  ch4_algo_parameters_container_in);
-    mpi_errno = MPIDI_OFI_Barrier_call(comm_ptr, errflag, nm_algo_parameters_container_out);
+
+    switch (nm_algo_parameters_container_out->id) {
+        case MPIDI_OFI_Barrier_intra_dissemination_id:
+            mpi_errno =
+                MPIDI_OFI_Barrier_intra_dissemination(comm_ptr, errflag,
```

Comment:
I'm confused about why we're duplicating all of the `MPIR` algorithms here. Can't we just call them directly instead of using these indirection functions?

Reply:
@wesbland @ddurnov  Till we don't have ofi specific implementations these functions are wrappers over MPIR algorithms. The current approach represents the skeleton of layered scheme for collective selection.

Reply:
I understand that there will be more in the future, but it seems list a ton of unnecessary code just to wrap the `MPIR` functions.

Reply:
Yeah, I understand what you're doing here. I just don't know about adding the placeholder. What do you think @pavanbalaji ?

Reply:
I agree with @wesbland.  Adding the placeholder function is weird.